lib/sync/validator.go

```diff
@@ -154,6 +160,15 @@ func (v *BlockValidator) validateBlock(ctx context.Context, si *SyncInfo) error
 	for _, tx := range si.Txs {
 		txs = append(txs, tx.H.Hash)
 	}
+
+	prevBlk, err := v.getPrevBlock(ctx, si.BlockHeight)
+	if err != nil {
+		return err
+	}
+
+	round := si.Block.Round
+	round.BlockHash = prevBlk.Hash
```

Comment:
How about this?
`si.Block.Round.BlockHash = prevBlk.Hash`
First I saw `round` variable, it looked like unused :)